guide/src/python_from_rust.md

```diff
@@ -236,6 +236,42 @@ def leaky_relu(x, slope=0.01):
 # }
 ```
 
+### Want to embed Python in Rust with exposed modules?
+
+Internally Python maintains the `sys.modules` dict.
+An import in Python is simply a lookup into this dict.
+
+In pyo3 it is nessercary to insert the new module into the dict
+before it is exposed to Python.
+
+```rust
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+#[pyfunction]
+pub fn add_one(x: i64) -> i64 {
+    x + 1
+}
+
+fn main() -> PyResult<()> {
+    Python::with_gil(|py| {
+        // Create new module
+        let foo_module = PyModule::new(py, "foo")?;
+        foo_module.add_function(wrap_pyfunction!(add_one, foo_module)?)?;
+
+        // Import and get sys.modules
+        let sys = PyModule::import(py, "sys")?;
+        let py_modules: &PyDict = sys.getattr("modules")?.downcast()?;
+
+        // Insert foo into sys.modules
+        py_modules.set_item("foo", foo_module)?;
```

Comment:
There is actually also another approach, which is to use a normal `#[pymodule] fn my_module` and `append_to_inittab!(my_module)`. Unfortunately the [docs for `append_to_inittab`](https://docs.rs/pyo3/latest/pyo3/macro.append_to_inittab.html) are woeful and lack an example.

It should be enough to call `append_to_inittab!(my_module)` _before_ calling `Python::with_gil`. Would you be willing to also document this option too?

Reply:
Job done.

One question - most of the links in `python_from_rust.md` use the following prefix 
`https://docs.rs/pyo3/0.12`. Is there a reason for using 0.12 instead of latest?

`append_to_inittab` isn't avaliable for 0.12 so I just used /latest/

Reply:
Looks great 👍 

Ah, for the docs URL, could you actually use `{{#PYO3_DOCS_URL}}` like [here in `class.md`](https://github.com/PyO3/pyo3/blob/61fd70c4764958898744dff90fadf2cc64851015/guide/src/class.md?plain=1#L1014) please? If you'd be willing to update the others to use that too that would be perfect, they shouldn't have hard-coded 0.12, I must have missed them when I originally built the substitution.

Reply:
Job done.

I haven't changed any of the other links in the file. Nor have I linked to the `append_to_inittab` example from the api reference (by api reference, I mean cargo doc). This seems to me to be out of scope for this PR?